interactors/src/main/java/com/thefuntasty/interactors/disposables/SingleDisposablesOwner.kt

```diff
@@ -13,8 +13,40 @@ interface SingleDisposablesOwner {
         return execute(args, { })
     }
 
-    fun <ARGS, T> BaseSingler<ARGS, T>.execute(args: ARGS): Disposable {
-        return execute(args, { })
+    fun <T> BaseSingler<Unit, T>.execute(result: SinglerResult.Builder<T>.() -> Unit): Disposable {
+        val singlerResult = SinglerResult.Builder<T>().run {
+            result.invoke(this)
+            return@run build()
+        }
+
+        if (singlerResult.disposePrevious) {
+            this@execute.currentDisposable?.dispose()
+        }
+
+        val disposable = create(Unit).subscribe(singlerResult.onSuccess, singlerResult.onError)
+
+        this@execute.currentDisposable = disposable
+        disposables += disposable
+
+        return disposable
+    }
+
+    fun <T> BaseSingler<Unit, T>.executeNoArgs(result: SinglerResult.Builder<T>.() -> Unit): Disposable {
```

Comment:
What is the difference between `execute` with no arguments and `executeNoArgs`? I think that the second is unnecessary.

Reply:
This is intentional as we can decide which one to use, the other will be removed.